src/rsa.py

```diff
@@ -80,45 +80,57 @@ def sign(body, user):
-            if cmd_result.returncode != 0:
-                raise RuntimeError("Failed to sign payload")
-            tmpout = tempfile.mkstemp(prefix="rsa-")[1]
-            with open(tmpout, 'a', encoding='utf-8') as output:
-                output.write(sha256sum(tmpinput))
-                output.write("\n")
-                output.write("ts: " +
+        with tempfile.NamedTemporaryFile(prefix="rsa-", delete=False,
+                                         mode='wb') as tmpinput:
+            tmpinput.write(payload)
+            tmpinput_path = tmpinput.name
+
+        try:
+            with tempfile.NamedTemporaryFile(prefix="rsa-", delete=False,
+                                             mode='wb') as tmpfile:
+                tmpfile_path = tmpfile.name
+
+                cmd = [ "openssl", "dgst", "-sign",
+                       os.path.join(RSA_DIR, key_id, "private.pem"),
+                       "-keyform", "PEM",
+                       "-sha256", "-out", tmpfile_path, tmpinput_path ]
+                subprocess.run(cmd, check=True)
+
+            with tempfile.NamedTemporaryFile(prefix="rsa-", delete=False,
+                                             mode='a', encoding='utf-8') as tmpout:
+                tmpout_path = tmpout.name
+                tmpout.write(sha256sum(tmpinput_path))
+                tmpout.write("\n")
+                tmpout.write("ts: " +
                              str(int(datetime.utcnow().timestamp())))
-                output.write("\n")
+                tmpout.write("\n")
                 hex_string=""
-                with open(tmpfile, 'rb') as ifile:
+                with open(tmpfile_path, 'rb') as ifile:
                     while True:
                         chunk = ifile.read(4096)
                         if not chunk:
                             break
                         hex_string += binascii.hexlify(chunk).decode('utf-8')
-                output.write("rsa2048: " + hex_string)
-                output.write("\n")
+                tmpout.write("rsa2048: " + hex_string)
+                tmpout.write("\n")
 
-            with open(tmpout, 'rb') as signature:
+            with open(tmpout_path, 'rb') as signature:
                 response = {
                     "signature": binascii.b2a_base64(signature.read()).decode().rstrip("\n")
                 }
 
                 LOG.info("%s successfully signed a payload using %s key", user, key_id)
-
-        os.remove(tmpinput)
-        os.remove(tmpout)
-        os.remove(tmpfile)
-        return response
+                return response
+        except subprocess.CalledProcessError as excp:
+            LOG.error("Subprocess failed: %s", excp)
+            return {"error": "Failed to sign payload"}, 500
+        except Exception as excp:
+            LOG.error("Unexpected error: %s", excp)
+            return {"error": str(excp)}, 500
+        finally:
+            os.remove(tmpinput_path)
```

Comment:
Each of these can be undefined based on where the exception is thrown. There are two ways around:
* Set each to `None` at the very top and check for `None` before removing
* Wrap each `os.remove` to `try/catch` and ignore errors

Both work, I personally prefer the first, it feels cleaner and the code would be more readable.